src/webportal/src/app/job/job-view/fabric/job-detail/components/task-role-container-list.jsx

```diff
@@ -218,40 +218,73 @@ export default class TaskRoleContainerList extends React.Component {
-      res.push(
-        'Solution 2: Use your own image, but enable SSH for it. Please follow the instructions on https://aka.ms/AA5u4sq to do such work.',
-      );
-      this.setState({
-        monacoProps: { value: res.join('\n') },
-        monacoTitle: `SSH to ${id}`,
-      });
+    if (config.launcherType !== 'k8s') {
+      if (!containerSshInfo) {
+        const res = [];
+        res.push('This job does not contain SSH info.');
+        res.push(
+          'Please note that if your docker image does not have openssh-server and curl packages, SSH will not be enabled.\n',
+        );
+        res.push(
+          'Solution 1: Use one of the recommended docker images on the submission page.',
+        );
+        res.push(
+          'Solution 2: Use your own image, but enable SSH for it. Please follow the instructions on https://aka.ms/AA5u4sq to do such work.',
+        );
+        this.setState({
+          monacoProps: { value: res.join('\n') },
+          monacoTitle: `SSH to ${id}`,
+        });
+      } else {
+        const res = [];
+        res.push('# Step 1. Open a Bash shell terminal.');
+        res.push('# Step 2: Download the private key:');
+        res.push(
+          `wget '${sshInfo.keyPair.privateKeyDirectDownloadLink}' -O ${sshInfo.keyPair.privateKeyFileName}`,
+        );
+        res.push('# Step 3: Set correct permission for the key file:');
+        res.push(`chmod 600 ${sshInfo.keyPair.privateKeyFileName}`);
+        res.push('# Step 4: Connect to the container:');
+        res.push(
+          `ssh -i ${sshInfo.keyPair.privateKeyFileName} -p ${containerSshInfo.sshPort} root@${containerSshInfo.sshIp}`,
+        );
+        res.push('');
+        this.setState({
+          monacoProps: {
+            value: res.join('\n'),
+            options: {
+              wordWrap: 'off',
+              readOnly: true,
+            },
+          },
+          monacoTitle: `SSH to ${id}`,
+        });
+      }
     } else {
       const res = [];
-      res.push('# Step 1. Open a Bash shell terminal.');
-      res.push('# Step 2: Download the private key:');
-      res.push(
-        `wget '${sshInfo.keyPair.privateKeyDirectDownloadLink}' -O ${sshInfo.keyPair.privateKeyFileName}`,
-      );
-      res.push('# Step 3: Set correct permission for the key file:');
-      res.push(`chmod 600 ${sshInfo.keyPair.privateKeyFileName}`);
-      res.push('# Step 4: Connect to the container:');
-      res.push(
-        `ssh -i ${sshInfo.keyPair.privateKeyFileName} -p ${containerSshInfo.sshPort} root@${containerSshInfo.sshIp}`,
-      );
-      res.push('');
+      if ('ssh' in containerPorts) {
+        res.push(
+          'You can connect to this container by one of the following commands if SSH is set up properly: \n',
+        );
+        res.push(`1. Use your default SSH private key:\n`);
+        res.push(`ssh -p ${containerPorts.ssh} root@${containerIp}\n`);
+        res.push(`2. Use a pre-downloaded SSH private key:\n`);
```

Comment:
add change permission command before ssh?

Reply:
Since it is not necessary to use `chmod` on Windows, I change the hint to:

```
On Windows:
ssh -p 20500 -i <your-private-key-file-path> root@10.a.b.c

On Unix-like System:
chmod 400 <your-private-key-file-path> && ssh -p 20500 -i <your-private-key-file-path> root@10.a.b.c
```

Reply:
change to 600?

Reply:
> change to 600?

Both OK. 400 is recomended because private id should not be changed.

Reply:
but previous code in this function still uses 600

Reply:
> but previous code in this function still uses 600, better to be consistent

Sure. I have modified previous code from 600 to 400.